db/compaction/compaction_job.cc

```diff
@@ -1340,8 +847,8 @@ void CompactionJob::ProcessKeyValueCompaction(SubcompactionState* sub_compact) {
 
   NotifyOnSubcompactionBegin(sub_compact);
 
-  CompactionRangeDelAggregator range_del_agg(&cfd->internal_comparator(),
-                                             existing_snapshots_);
+  auto range_del_agg = std::make_unique<CompactionRangeDelAggregator>(
```

Comment:
I think it's recommended to use `auto` with `make_unique<TypeClass>()`, as we specified the type in `make_unique`, to avoid repeat the TypeClass twice in one sentence: https://google.github.io/styleguide/cppguide.html#Type_deduction